services/cognito/app/controllers/metabase_token_controller.rb

```diff
@@ -29,4 +31,18 @@ def payload_params
 
     options.merge(params[:params].to_unsafe_h.deep_symbolize_keys)
   end
+
+  def map_identifier_to_id
+    return if params[:identifier].blank?
+
+    if card_map.card_identifier_record.present?
+      params[:id] = card_map.mapped_value
+    else
+      render json: { errors: 'Card ID not found' }
+    end
+  end
+
+  def card_map
+    Metabase::CardMap.new(identifier: params[:identifier])
```

Comment:
Creating this seems like a slight overkill. In the end we just need the id for the card. its a simple find_by